pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -326,6 +327,22 @@ func (c *Controller) execute(key string) error {
 		return err
 	}
 
+	if pod != nil {
+		_, podIsMarkedForEviction := pod.GetAnnotations()[descheduler.EvictionInProgressAnnotation]
+		if vmi.IsMarkedForEviction() && !podIsMarkedForEviction {
+			pod, err = descheduler.MarkEvictionInProgress(c.clientset, pod)
+			if err != nil {
+				return err
+			}
+		}
+		if !vmi.IsMarkedForEviction() && podIsMarkedForEviction {
+			pod, err = descheduler.MarkEvictionCompleted(c.clientset, pod)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
I am not entirely sure this is the right place to do this. I'd expect that this should be under the `sync` or the `updateStatus`. Is there a specific reason having this here?